### PR TITLE
Provide partial EMR support

### PR DESCRIPTION
**Pull request by mrallen1**
This is the beginning of Elastic Map Reduce support. Currently the four API calls which are supported are `AddJobStep`, `RunJobFlow`, `TerminateJobFlow`, and `DescribeSteps`.  This is the "minimum viable" amount of support needed to use EMR.

This PR supersedes #348 - since that PR, the code has been rebased against current master and a test added, as per comments.

### CODE REVIEW

**Comment by nalundgaard** [comment]
Please make sure and add this to [`erlcloud_aws:service_config/3` and/or `erlcloud_aws:service_host/2`](https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_aws.erl#L495-L632) as appropriate.

File: include/erlcloud_aws.hrl [hunk 1]
```diff
@@ -30,6 +30,9 @@
           sqs_host="queue.amazonaws.com"::string(),
           sqs_protocol=undefined::string()|undefined,
           sqs_port=undefined::non_neg_integer()|undefined,
+          emr_scheme="https://"::string()|undefined,
+          emr_host="elasticmapreduce.amazonaws.com"::string(),
```
Comment (nalundgaard) [inline include/erlcloud_aws.hrl]: should this be `elasticmapreduce.us-east-1.amazonaws.com`?

**Comment by mrallen1** [comment]
How's this look?

**Comment by nalundgaard** [comment]
Am I reading correctly that the tests simply cover the common request mechanics and the run_job_flow function? Any reason to not have I/O tests for the other added functions?

**Comment by mrallen1** [comment]
You're reading correctly. 

I don't understand the utility of adding more mocked tests. At best they are ensuring that jsx can serialize and deserialize JSON correctly. At worst they are a false sense of security that real-world Amazon services behave as the mocked HTTP calls.

Are there tests that do not involve mocks and make live service calls? I'd be happy to write some of those for EMR.

**Comment by mrallen1** [comment]
OK, I mocked out the functions which have example input/output data in the API docs. I think that's all the tests which are feasible given the docs.

File: test/erlcloud_emr_tests.erl [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+     [{<<"Jar">>,
+       <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/cloudburst.jar">>},
+      {<<"Args">>,
+       [<<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/s_suis.br">>,
+        <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/100k.br">>,
+        <<"s3n:\\/\\/examples-bucket\\/cloudburst\\/output">>,<<"36">>,
+        <<"3">>,<<"0">>,<<"1">>,<<"240">>,<<"48">>,<<"24">>,<<"24">>,
+        <<"128">>,<<"16">>]}]}]]}]).
+
+%% AddJobFlowSteps doesn't have an example output format.
```
Comment (nalundgaard) [inline test/erlcloud_emr_tests.erl]: that is wild, it looks like there is a response syntax that includes [StepIds](http://docs.aws.amazon.com/ElasticMapReduce/latest/API/API_AddJobFlowSteps.html#API_AddJobFlowSteps_ResponseSyntax). I guess their docs guy fell asleep at the wheel.

Comment (nalundgaard) [inline test/erlcloud_emr_tests.erl]: Would it be reasonable to expect output like
```json
{"JobFlowIds": ["j-3TS0OIYO4NFN"]}
```
based on the doc?

**Comment by mrallen1** [comment]
Really making me work for it. OK, how about now?

**Comment by nalundgaard** [comment]
👍 LGTM. @motobob or @waisbrot any objections? If not, I'll merge this later today.

**Pull request merged**